src/components/MdDialog/MdDialog.vue

```diff
@@ -17,205 +28,223 @@
-  import MdPortal from 'components/MdPortal/MdPortal'
-  import MdOverlay from 'components/MdOverlay/MdOverlay'
-  import MdFocusTrap from 'components/MdFocusTrap/MdFocusTrap'
-
-  export default new MdComponent({
-    name: 'MdDialog',
-    components: {
-      MdPortal,
-      MdOverlay,
-      MdFocusTrap
-    },
-    props: {
-      mdActive: Boolean,
-      mdBackdrop: {
-        type: Boolean,
-        default: true
-      },
-      mdBackdropClass: {
-        type: String,
-        default: 'md-dialog-overlay'
-      },
-      mdCloseOnEsc: {
-        type: Boolean,
-        default: true
-      },
-      mdClickOutsideToClose: {
-        type: Boolean,
-        default: true
-      },
-      mdFullscreen: {
-        type: Boolean,
-        default: true
-      },
-      mdAnimateFromSource: Boolean
-    },
-    computed: {
-      dialogClasses () {
-        return {
-          'md-active': this.mdActive
-        }
-      },
-      dialogContainerClasses () {
-        return {
-          'md-dialog-fullscreen': this.mdFullscreen
-        }
-      }
-    },
-    watch: {
-      mdActive (isActive) {
-        this.$nextTick().then(() => {
-          if (isActive) {
-            this.$emit('md-opened')
-          } else {
-            this.$emit('md-closed')
-          }
-        })
-      }
-    },
-    methods: {
-      closeDialog () {
-        this.$emit('update:mdActive', false)
-      },
-      onClick () {
-        if (this.mdClickOutsideToClose) {
-          this.closeDialog()
-        }
-        this.$emit('md-clicked-outside')
-      },
-      onEsc () {
-        if (this.mdCloseOnEsc) {
-          this.closeDialog()
-        }
-      }
-    }
-  })
+	import MdComponent from "core/MdComponent";
+	import MdPortal from "components/MdPortal/MdPortal";
+	import MdOverlay from "components/MdOverlay/MdOverlay";
+	import MdFocusTrap from "components/MdFocusTrap/MdFocusTrap";
+
+	export default new MdComponent({
+		name: "MdDialog",
+		components: {
+			MdPortal,
+			MdOverlay,
+			MdFocusTrap,
+		},
+		props: {
+			mdActive: Boolean,
+			mdTarget: {
+				type: null,
+				default: null,
+			},
+			mdBackdrop: {
+				type: Boolean,
+				default: true,
+			},
+			mdBackdropClass: {
+				type: String,
+				default: "md-dialog-overlay",
+			},
+			mdCloseOnEsc: {
+				type: Boolean,
+				default: true,
+			},
+			mdClickOutsideToClose: {
+				type: Boolean,
+				default: true,
+			},
+			mdFullscreen: {
+				type: Boolean,
+				default: true,
+			},
+			mdAnimateFromSource: Boolean,
+		},
+		computed: {
+			dialogClasses() {
+				return {
+					"md-active": this.mdActive,
+				};
+			},
+			dialogContainerClasses() {
+				return {
+					"md-dialog-fullscreen": this.mdFullscreen,
+				};
+			},
+		},
+		watch: {
+			mdActive(isActive) {
+				if (this.mdTarget) {
+					if (isActive) {
+						this.mdTarget.style.overflow = "hidden";
+					} else {
+						this.mdTarget.style.overflow = "auto";
+					}
+				}
```

Comment:
forcing overflow behavioiur on target element if present to set the overlay fixed